backend/app/processing/vanna_integrator.py

```
@@ -0,0 +1,267 @@
+from pathlib import Path
+import sqlite3
+from typing import Dict, Any
+import os
+import shutil
+from core.logger import logger
+
+# Use your custom subclass that combines ChromaDB vector store + Gemini chat
+from vanna.chromadb import ChromaDB_VectorStore
+from vanna.google import GoogleGeminiChat
+# PersistentClient to make Chroma auto‑persist into our folder
```

Ruff (0.8.2)

Comment contains ambiguous `‑` (NON-BREAKING HYPHEN). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)